### PR TITLE
Issue #1058 make minikube iso usable for minishift

### LINKED ISSUE #1058: Add support for Minikube ISO
Currently we are only supporting b2d and centOS, we should also check minikube iso which is based on buildroot and have docker + rkt runtime.

### related pulls
  * [x] bind mount https://github.com/kubernetes/minikube/pull/1968
  * [x] Minishift paths https://github.com/kubernetes/minikube/pull/1993

### PR DESCRIPTION
**Pull request by praveenkumar**
Fix #1058 

- We still need to update the iso version once it released along with minishift directory structure.  : Done

### CODE REVIEW

File: pkg/minishift/provisioner/buildroot_provisioner.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
```
Comment (budhrg) [inline pkg/minishift/provisioner/buildroot_provisioner.go]: Shouldn't it be Red Hat license?

File: pkg/minishift/provisioner/buildroot_provisioner_test.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
+/*
+Copyright (C) 2016 Red Hat, Inc.
```
Comment (budhrg) [inline pkg/minishift/provisioner/buildroot_provisioner_test.go]: This is correct though year is incorrect.

File: pkg/minishift/provisioner/utils.go [hunk 3]
```diff
@@ -58,9 +59,53 @@ ExecStart=/usr/bin/dockerd-current -H tcp://0.0.0.0:{{.DockerPort}} -H unix:///v
...
+
+# kill only the docker process, not all processes in the cgroup
+KillMode=process
+
+[Install]
+WantedBy=multi-user.target`
 )
 
-func makeDockerOptionsDir(p *MinishiftProvisioner) error {
+func makeDockerOptionsDir(p provision.Provisioner) error {
```
Comment (budhrg) [inline pkg/minishift/provisioner/utils.go]: :+1: , generic now.

File: pkg/minishift/provisioner/buildroot_provisioner.go (hunk 1 above)
Comment (gbraad) [reply pkg/minishift/provisioner/buildroot_provisioner.go]: I think this is partially based on https://github.com/kubernetes/minikube/blob/master/pkg/provision/buildroot.go

**Comment by praveenkumar** [comment]
Need to do some more changes around buildroot provisioner side.

File: pkg/minikube/assets/vm_assets.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+		return errors.Wrapf(err, "error changing file permissions for %s", targetPath)
+	}
+
+	if _, err = io.Copy(target, f); err != nil {
+		return errors.Wrapf(err, `error copying file %s to target location:
+do you have the correct permissions?  The none driver requires sudo for the "start" command`,
+			targetPath)
+	}
+	return target.Close()
+}
```
Comment (praveenkumar) [inline pkg/minikube/assets/vm_assets.go]: This file is coming from https://github.com/kubernetes/minikube/blob/d421074c697f2e2fa03df8e276ac66623dc69dcb/pkg/minikube/assets/vm_assets.go

File: pkg/minikube/sshutil/sshutil.go [hunk 5]
```diff
@@ -61,6 +63,12 @@ func NewSSHClient(d drivers.Driver) (*ssh.Client, error) {
 	return client, nil
 }
 
+func TransferFile(f assets.CopyableFile, client *ssh.Client) error {
+	return Transfer(f, f.GetLength(),
+		f.GetTargetDir(), f.GetTargetName(),
+		f.GetPermissions(), client)
+}
+
```
Comment (praveenkumar) [inline pkg/minikube/sshutil/sshutil.go]: This is again updated from https://github.com/kubernetes/minikube/blob/2d2b484a22c389d257b6745675c29ddd264bb3d3/pkg/minikube/sshutil/sshutil.go#L95-L100

File: pkg/minishift/provisioner/buildroot_provisioner.go (hunk 1 above)
Comment (praveenkumar) [reply pkg/minishift/provisioner/buildroot_provisioner.go]: yes it's based on kubernetes buildroot provision

File: pkg/minishift/provisioner/utils.go [hunk 6]
```diff
@@ -121,3 +174,94 @@ func getFilesystemType(p *MinishiftProvisioner, directory string) (string, error
...
+	if _, err = p.SSHCommand(fmt.Sprintf("sudo mkdir -p %s && printf %%s \"%s\" | sudo tee %s", path.Dir(dockerCfg.EngineOptionsPath), dockerCfg.EngineOptions, dockerCfg.EngineOptionsPath)); err != nil {
+		return err
+	}
+
+	if err := p.Service("docker", serviceaction.Restart); err != nil {
+		return err
+	}
+
+	return nil
+}
```
Comment (praveenkumar) [inline pkg/minishift/provisioner/utils.go]: This is coming from https://github.com/kubernetes/minikube/blob/9b2dd165b375c2db0d7b2755dd63f60f359e30d0/pkg/provision/buildroot.go#L181-L270

**Comment by gbraad** [comment]
'useable' => 'usable'

**Comment by gbraad** [comment]
retest this please

**Comment by gbraad** [comment]
retest this please

**Comment by budhrg** [comment]
retest this please

**Comment by gbraad** [comment]
Still an issue with: continuous-integration/appveyor/pr — Waiting for status to be reported ?

**Comment by gbraad** [comment]
Merged with a4373c0ab35b5eea77a4dca5783bfe36bc593e18

**Pull request closed without merging**